### ISSUE TITLE
Support deprecation

### ISSUE THREAD

**Issue by nickevansuk** [openactive/data-model-validator#368]
## 1. Deprecation of model properties

Based on https://github.com/openactive/modelling-opportunity-data/issues/221

For any fields that are used that have the `deprecationGuidance` property set:
- WARNING in validation mode `RPDEFeed` (open data is more forgiving than the API)
- FAILURE if the validation mode is anything other than `RPDEFeed`

The `deprecationGuidance` string is the error message that is displayed

e.g. https://github.com/openactive/data-models/blob/f3afdb2485fe07ddaac1550c58eb6037eaf29b67/versions/2.x/models/Offer.json#L163-L171

```json
    "ageRange": {
      "fieldName": "ageRange",
      "sameAs": "https://openactive.io/ageRange",
      "model": "#QuantitativeValue",
      "description": [
        "Indicates that an Offer is only applicable to a specific age range."
      ],
      "deprecationGuidance": "Use `ageRestriction` instead of `ageRange` within the `Offer` for cases where the `Offer` is age restricted."
    },
```

See documentation here: https://github.com/openactive/data-models/blob/master/MODELS.md#deprecationguidance

## 2. Deprecation of extension properties

Use [`beta:timeZone`](https://openactive.io/ns-beta#timeZone) (inside `Schedule`) to test extension deprecation. This is the first of many beta properties to be deprecated.

Support for deprecation of beta (extension) can be added by leveraging the [`supersededBy` property that exists in the namespace file](https://github.com/openactive/ns-beta/blob/4d12387c5e4f135a1458c1c8481b0c57854b0da5/beta.jsonld#L142).

Add to the `GraphHelper`; use `this.getProperty()` to also retrieve the `schema:supersededBy`, and pass it back inside the existing successful `return this.isPropertyInClassReturn(GraphHelper.PROPERTY_FOUND, {supersededBy} )`:
https://github.com/openactive/data-model-validator/blob/83115aef88bf8ec3a3fe40145ac5e589e530c628/src/helpers/graph.js#L135-L135

Then check for `graphResponse.data.supersededBy` within this if statement:
https://github.com/openactive/data-model-validator/blob/83115aef88bf8ec3a3fe40145ac5e589e530c628/src/rules/core/fields-not-in-model-rule.js#L328-L331

If `graphResponse.data.supersededBy` exists, use error message similar to [this one](https://github.com/openactive/models-lib/blob/5f4ec20ec4cf2e1020babad662311e02fd125ab6/src/generator.js#L608-L608) for deprecated properties, as follows:

Error message: "This term has graduated from the beta namespace and is now deprecated, please use `{supersededBy}` instead."
- WARNING in validation mode `RPDEFeed` (open data is more forgiving than the API)
- FAILURE if the validation mode is anything other than `RPDEFeed`

**Comment by nickevansuk** [comment]
### Implementation note

The should be implemented a more general rule, leveraging the recently added `deprecated` and `deprecationGuidance` properties in the data models rather than specifically hardcoding `ageRange`:

https://github.com/openactive/data-models/commit/7737e810036f7fbfe7cb15ea50f0cbf69afac640#diff-acf4988cb1eb345a30fdcdbccc1cea60a56873ffbc6b5fc31ea2b4994933d3c2R461-R467

**Comment by nickevansuk** [comment]
Closed by #411
